src/test/java/io/personium/core/bar/BarFileExporterTest.java

```diff
@@ -0,0 +1,322 @@
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void prepareBarFile() {
+        // prepare mocks for testing
+        Cell cell = mockCell(CELL_URL);
+        Box box = mockBox(cell, BOX_NAME, BOX_SCHEMA_URL);
+        BoxRsCmp boxRsCmpMock = mockBoxRsComp(box);
+
+        // create bar file response
+        BarFileExporter exporter =  new BarFileExporter(boxRsCmpMock);
+        //  ... this part is the core of this unit test ...
+        Response res = exporter.export();
+
+        // parse the response bar file contents
+        StreamingOutput so = (StreamingOutput)res.getEntity();
+        byte[] barFileBytes = responseToBytes(so);
+        readExportedBarFile(barFileBytes);
+    }
+
+    @Before
+    public void before() throws Exception {
+    }
+
+    /**
+     * bar file composition check.
+     */
+    @Test
+    public void export_BarFile_ShouldInclude_NecessaryZipEntries() {
+        ZipEntry meta = barZipEntryMap.get("00_meta/");
+        assertNotNull(meta);
+        assertTrue(meta.isDirectory());
+
+        ZipEntry manifest = barZipEntryMap.get("00_meta/00_manifest.json");
+        assertNotNull(manifest);
+        assertFalse(manifest.isDirectory());
+
+        ZipEntry rootprops = barZipEntryMap.get("00_meta/90_rootprops.xml");
+        assertNotNull(rootprops);
+        assertFalse(rootprops.isDirectory());
+
+        ZipEntry contents = barZipEntryMap.get("90_contents/");
+        assertNotNull(contents);
+        assertTrue(contents.isDirectory());
+
+        ZipEntry col = barZipEntryMap.get("90_contents/col/");
+        assertNotNull(col);
+        assertTrue(col.isDirectory());
+
+        ZipEntry f1 = barZipEntryMap.get("90_contents/col/file1.json");
+        assertNotNull(f1);
+        assertFalse(f1.isDirectory());
+
+        ZipEntry f2 = barZipEntryMap.get("90_contents/file2.json");
+        assertNotNull(f2);
+        assertFalse(f2.isDirectory());
+
+    }
+
+    /**
+     * check for 90_rootprops.xml in the exported bar file.
+     * @throws Exception
+     */
+    @Test
+    public void export_RootpropsXml_ShouldHave_ValidContents() throws Exception {
+        byte[] b = barZipContentMap.get("00_meta/90_rootprops.xml");
+        String rootpropsXml = new String(b);
+        log.info("00_meta/90_rootprops.xml\n----\n" + rootpropsXml + "\n----");
+
+        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+        DocumentBuilder db = dbf.newDocumentBuilder();
+
+        Document doc = db.parse(new InputSource(new StringReader(rootpropsXml)));
+        XPath xpath = XPathFactory.newInstance().newXPath();
+
+        // ACL base Url should be the base url of Role Class Url
+        String base = xpath.evaluate("//acl[position()=1]/@base", doc);
+        log.info("//acl[position()=1]/@base = " + base);
+        assertEquals(BOX_SCHEMA_URL + "__role/__/", base);
```

Comment:
Just a question. Is Role class URL `${schema URL}/__role/__/${RoleName}` or `${schema URL}/__role/__/` ?

https://personium.io/docs/ja/user_guide/008_Glossary.html#anc_r

says `${schema URL}/__role/__/${RoleName}`. Since this path is base url,  `${schema URL}/__role/__/` is used?

Reply:
The above document is correct. Role class URL is something like
````
  ${schema URL}/__role/__/${RoleName}
````

"base" attribute in the xml is for giving base url, so that the URLs in the href elements can be written in relative URL.
````
 href:   someRole (relative URL)
 base:  https://app1.pds.example/__role/__/
````

Then href is interpreted as 
````
https://app1.pds.example/__role/__/someRole
````